aikau/src/main/resources/alfresco/lists/ItemSelectionMixin.js

```diff
@@ -0,0 +1,252 @@
+               }
+            }
+            else if (payload.selectedItems)
+            {
+               // Check to see if the list of selected items contains the item that this instance has
+               // renderered.
+               array.some(payload.selectedItems, function(item) {
+                  var a = lang.getObject(this.itemKey, false, item);
+                  var b = lang.getObject(this.itemKey, false, this.currentItem);
+                  var match = ((a || a === 0) && a === b);
+                  if (match) {
+                     domClass.add(this.getSelectorNode(), "alfresco-lists-ItemSelectionMixin--checked");
+                  }
+                  return match;
+               }, this);
+            }
+         }
+      },
+      
+      /**
+       * Updates the CSS classes to indicate that the item has been selected and publishes 
+       * [the topic]{@link module:alfresco/documentlibrary/_AlfDocumentListTopicMixin#documentSelectedTopic}
+       * to indicate that an item has been selected.
+       * 
+       * @instance
+       * @fires module:alfresco/documentlibrary/_AlfDocumentListTopicMixin#documentSelectedTopic
+       */
+      select: function alfresco_lists_ItemSelectionMixin__select() {
+         domClass.add(this.getSelectorNode(), "alfresco-lists-ItemSelectionMixin--checked");
+         this.alfPublish(this.documentSelectedTopic, {
+            value: this.currentItem
+         }, this.getSelectionPublishGlobal(), this.getSelectionPublishToParent());
+      },
+      
+      /**
+       * Updates the CSS classes to indicate that the item has been de-selected and publishes 
+       * [the topic]{@link module:alfresco/documentlibrary/_AlfDocumentListTopicMixin#documentDeselectedTopic}
+       * to indicate that an item has been de-selected.
+       * 
+       * @instance
+       * @fires module:alfresco/documentlibrary/_AlfDocumentListTopicMixin#documentDeselectedTopic
+       */
+      deselect: function alfresco_lists_ItemSelectionMixin__deselect() {
+         domClass.remove(this.getSelectorNode(), "alfresco-lists-ItemSelectionMixin--checked");
+         this.alfPublish(this.documentDeselectedTopic, {
+            value: this.currentItem
+         }, this.getSelectionPublishGlobal(), this.getSelectionPublishToParent());
+      },
+      
+      /**
+       * If [selectOnClick]{@link module:alfresco/lists/ItemSelectionMixin#selectOnClick} is configured to be
+       * true then this will call either [deselect]{@link module:alfresco/lists/ItemSelectionMixin#deselect}
+       * or [select]{@link module:alfresco/lists/ItemSelectionMixin#select} (depending upon the current
+       * state).
+       * 
+       * @instance
+       */
+      onSelectionClick: function alfresco_lists_ItemSelectionMixin__onSelectionClick(evt) {
+         if (this.selectOnClick)
+         {
+            evt && Event.stop(evt);
+            if (domClass.contains(this.getSelectorNode(), "alfresco-lists-ItemSelectionMixin--checked"))
```

Comment:
No, because they existed previously in Selector (which now mixes in this module). If someone has extended Selector they maybe calling, overriding or extending select and/or deslect .